tools/ci/prepare-pull-credentials.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+set -e
+
+if [[ "$DOCKERHUB_PULL_CREDENTIAL" == "" ]]; then
```

Comment:
Why can't this just be invoked as part of the makefile?

Reply:
You mean embeding this whole script file to the makefile `docker.mk` or `e2e.new.mk`? 
I think the yq expressions underneath are too complicated to be embeded.

Reply:
If you mean we can put the `if` check into the makefile, then yes, we can do it.